### PR TITLE
Implemented cross product of two vectors

### PR DESCRIPTION
**Pull request by hardik-yadav1729**
This program defines functions to calculate the cross product of two 3D vectors and the magnitude of a vector from its direction ratios.

I hope adding these programs would be helpful.

Thank you for consideration

### CODE REVIEW

File: src/math/vector_cross_product.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+/// Function to calculate the cross product of two vectors
+pub fn cross_product(vec1: [f64; 3], vec2: [f64; 3]) -> [f64; 3] {
+    let x = vec1[1] * vec2[2] - vec1[2] * vec2[1];
+    let y = -(vec1[0] * vec2[2] - vec1[2] * vec2[0]);
+    let z = vec1[0] * vec2[1] - vec1[1] * vec2[0];
+    [x, y, z]
+}
+
+/// Function to calculate the magnitude of a vector
+fn vector_magnitude(vec: [f64; 3]) -> f64 {
```
Comment (suzp1984) [inline src/math/vector_cross_product.rs:21]: do you forget `pub` this fn?

File: src/math/mod.rs [hunk 2]
```diff
@@ -54,6 +54,7 @@ mod sum_of_digits;
 mod tanh;
 mod trial_division;
 mod zellers_congruence_algorithm;
+pub mod vector_cross_product;
```
Comment (suzp1984) [inline src/math/mod.rs:57]: There are no reason to `pub` this mod.

**Comment by hardik-yadav1729** [comment]
I corrected these mistakes!
Thank you for pointing them out

File: src/math/vector_cross_product.rs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+/// Function to calculate the cross product of two vectors
+pub fn cross_product(vec1: [f64; 3], vec2: [f64; 3]) -> [f64; 3] {
+    let x = vec1[1] * vec2[2] - vec1[2] * vec2[1];
+    let y = -(vec1[0] * vec2[2] - vec1[2] * vec2[0]);
+    let z = vec1[0] * vec2[1] - vec1[1] * vec2[0];
+    [x, y, z]
+}
+
+/// Function to calculate the magnitude of a vector
+pub fn vector_magnitude(vec: [f64; 3]) -> f64 {
```
Comment (suzp1984) [inline src/math/vector_cross_product.rs:21]: I believe the fn also need to be public in the mod.rs

`pub use self::vector_cross_product::vector_magnitude;`

**Comment by hardik-yadav1729** [comment]
Yes , Thank you for pointing this out.

I made the required changes!

**Review by suzp1984 — APPROVED** [review]
good

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Comment by siriak** [comment]
Please check why the build is failing

**Comment by hardik-yadav1729** [comment]
I corrected the format.
please review

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**